src/App/App.js

```diff
@@ -38,13 +39,22 @@ class App extends Component {
   }
 
   logOutUser = () => {
-    this.setState({user: {}})
+    this.setState({user: {}, hasLoginView: false})
+  }
+
+  updateLoginView = () => {
+    if (!this.state.user.name) {
+      this.setState({hasLoginView: true})
+    }
   }
 
   determineLogButtonStatus = () => {
     if (this.state.user.name) {
       return <button className='log-button' onClick={this.logOutUser}>Logout</button>
-    } else {
+    } else if (this.state.hasLoginView) {
+      return 
```

Comment:
I do like how clean this return statement is, but I want to remember to ask about developer empathy